packages/mobile/src/account/Settings.tsx

```diff
@@ -258,9 +263,36 @@ export class Account extends React.Component<Props, State> {
     ValoraAnalytics.track(SettingsEvents.tos_view)
   }
 
+  onRemoveAccountPress = () => {
+    this.setState({ showAccountKeyModal: true })
+  }
+
+  hideRemoveAccountModal = () => {
+    this.setState({ showAccountKeyModal: false })
+  }
+
+  onPressContinueWithAccountRemoval = () => {
+    this.setState({ showAccountKeyModal: false })
+    this.props.navigation.navigate(Screens.BackupPhrase, { fromSettings: true })
```

Comment:
We're passing this `fromSettings` param through all the backup screens and the final one (`BackupComplete`) redirects to this `Settings` screen with the param `promptConfirmRemovalModal` as `true`.

Another option would be to store this on the redux store instead, but I thought it would be more error prone since we must make sure the flag is always reset if the user exits the flow at any point and it would be easy to forget/not realize we should do that when/if we make changes to this flow since it's not intuitive.